grapher/utils/Util.ts

```diff
@@ -1063,7 +1070,84 @@ export function sortNumeric<T>(
     return arr.sort(compareFn)
 }
 
+function isNotInvalidOrEmptyCell(value: any) {
+    return value !== undefined && !(value instanceof InvalidCell)
+}
+
 // https://github.com/robertmassaioli/ts-is-present
 // A predicate for filtering an array of nulls and undefineds that returns the correct type
 export const isPresent = <T>(t: T | undefined | null | void): t is T =>
     t !== undefined && t !== null
+
+export function fillUndefinedWithClosest<
+    ValueSlug extends ColumnSlug,
+    TimeSlug extends ColumnSlug,
+    Row extends { [key in TimeSlug]?: number } & { [key in ValueSlug]?: any }
+>(
+    rows: Row[],
```

Comment:
Looks good.

Does this assume rows are sorted by time asc, right? Maybe rename the `rows` argument to `rowsSortedByTimeAsc`?

Sometimes I've passed unsorted values into utils and gotten unexpected results.